### PR TITLE
iface docs: add method docs

### PR DESCRIPTION
**Pull request by robin-da**
This generates documentation for methods present in interface
definitions.

Not sure if we could also have doc strings added to the methods.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
thanks!

File: compiler/damlc/daml-doc/src/DA/Daml/Doc/Transform/Instances.hs [hunk 1]
```diff
@@ -76,3 +77,17 @@ distributeInstanceDocs opts docs =
...
+      { if_methods =
+          [ MethodDoc{..}
+          | InstanceDoc {id_type, id_module} <-
+                maybe [] Set.toList $ do
+                  anchor <- if_anchor idoc
+                  Map.lookup anchor imap
+          , Just "HasMethod" == getTypeAppName id_type
+          , "DA.Internal.Desugar" == id_module
+          , Just [_if_name, TypeLit name, mtd_type] <- [getTypeAppArgs id_type]
+          , let mtd_name = Typename $ T.filter (/= '"') name
```
[the remark this answers is not in the document]
Comment (robin-da) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Transform/Instances.hs:91]: the 'name' contains the leading and trailing double-quotes. we could also unpack and pattern match.

File: compiler/damlc/tests/daml-test-files/Interface.EXPECTED.md [hunk 2]
```diff
@@ -43,6 +43,18 @@
 >   | Field                                                                                   | Type                                                                                    | Description |
 >   | :-------------------------------------------------------------------------------------- | :-------------------------------------------------------------------------------------- | :---------- |
 >   | newOwner                                                                                | [Party](https://docs.daml.com/daml/stdlib/Prelude.html#type-da-internal-lf-party-50311) |  |
+> 
+> * **Method getAmount : **[Int](https://docs.daml.com/daml/stdlib/Prelude.html#type-ghc-types-int-68728)
```
[the remark this answers is not in the document]
Comment (robin-da) [reply compiler/damlc/tests/daml-test-files/Interface.EXPECTED.md:47]: yes I will do that in the next PR. I tried with `-- ^ doc string` but that leads to a parse error (same in template fields). did we somehow go with simple `-- doc string` for fields?

File: compiler/damlc/daml-doc/src/DA/Daml/Doc/Transform/Instances.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cocreature) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Transform/Instances.hs:91]: unpack & pattern match sounds nicer

**Pull request merged**